### PR TITLE
WIP:sidecar registry draft pr

### PR DESCRIPTION
**Pull request by thenicetgp**

### CODE REVIEW

File: pkg/adapter/sidecarprovider/sidecarprovider.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+import (
+	"github.com/apache/dubbo-go-pixiu/pkg/adapter/dubboregistry/registry"
+	"github.com/apache/dubbo-go-pixiu/pkg/common/constant"
+	"github.com/apache/dubbo-go-pixiu/pkg/common/extension/adapter"
+	"github.com/apache/dubbo-go-pixiu/pkg/logger"
+	"github.com/apache/dubbo-go-pixiu/pkg/model"
+	"github.com/apache/dubbo-go-pixiu/pkg/server"
+	"github.com/dubbo-go-pixiu/pixiu-api/pkg/api/config"
+	"github.com/dubbo-go-pixiu/pixiu-api/pkg/router"
```
Comment (baerwang) [inline pkg/adapter/sidecarprovider/sidecarprovider.go]: import style

File: pkg/adapter/sidecarprovider/sidecarprovider.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	// Plugin the adapter plugin for sidecar provider
+	Plugin struct {
+	}
+
+	// SidecarAdapter the adapter for sidecar register
+	SidecarAdapter struct {
+		id		  string
+		cfg       *AdaptorConfig
+		registries map[string]registry.Registry
+		URL		  *common.URL
```
Comment (baerwang) [inline pkg/adapter/sidecarprovider/sidecarprovider.go]: format

File: pkg/adapter/sidecarprovider/sidecarprovider.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+func (p *Plugin) CreateAdapter(a *model.Adapter) (adapter.Adapter, error) {
+	adapter := &SidecarAdapter{
+		cfg:      &AdaptorConfig{},
+	}
+	return adapter,nil
+}
+
+// Start starts the adaptor
+func (a SidecarAdapter) Start() {
+	//register sidecar
```
Comment (baerwang) [inline pkg/adapter/sidecarprovider/sidecarprovider.go]: space

File: pkg/adapter/sidecarprovider/sidecarprovider.go [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+	if url == nil {
+		return
+	}
+	instance, err := createInstance(url)
+	if err != nil {
+		panic(err)
+	}
+	p := extension.GetProtocol(dgconstant.RegistryKey)
+	var rp dgregistry.RegistryFactory
+	var ok bool
```
Comment (baerwang) [inline pkg/adapter/sidecarprovider/sidecarprovider.go]: use 
```go
var (
	rp dgregistry.RegistryFactory
	ok bool
)
```

File: pkg/filter/network/dubboproxy/filter/apiconfig/api_config.go [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package apiconfig
+
+import (
+	"context"
+	dubboConstant "dubbo.apache.org/dubbo-go/v3/common/constant"
+	"dubbo.apache.org/dubbo-go/v3/protocol/dubbo3"
```
Comment (baerwang) [inline pkg/filter/network/dubboproxy/filter/apiconfig/api_config.go]: import style

File: pkg/filter/network/dubboproxy/filter/apiconfig/api_config.go [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+import (
+	"github.com/apache/dubbo-go-pixiu/pkg/common/constant"
+	"github.com/apache/dubbo-go-pixiu/pkg/common/extension/filter"
+	"github.com/apache/dubbo-go-pixiu/pkg/config"
+	dubbo2 "github.com/apache/dubbo-go-pixiu/pkg/context/dubbo"
+	contexthttp "github.com/apache/dubbo-go-pixiu/pkg/context/http"
+	"github.com/apache/dubbo-go-pixiu/pkg/filter/network/dubboproxy/filter/apiconfig/api"
+	"github.com/apache/dubbo-go-pixiu/pkg/logger"
+	"github.com/apache/dubbo-go-pixiu/pkg/server"
+	fc "github.com/dubbo-go-pixiu/pixiu-api/pkg/api/config"
```
Comment (baerwang) [inline pkg/filter/network/dubboproxy/filter/apiconfig/api_config.go:45]: import style

Comment (thenicetgp) [reply pkg/filter/network/dubboproxy/filter/apiconfig/api_config.go:45]: Thanks for your review

**Pull request closed without merging**